packages/headless/src/composables/menuButton/createMenuButton.ts

```diff
@@ -142,7 +138,12 @@ export const createMenuButton = createBuilder((options: CreateMenuButtonOptions)
             "aria-controls": menuId,
             "aria-expanded": options.isExpanded.value,
             "aria-haspopup": true,
-            onFocus: () => setExpanded(true),
+            onFocus: () => setExpanded(true, true),
+            onClick: () => {
+              if (options.trigger.value == "click") {
+                setExpanded(!options.isExpanded.value);
+              }
+            },
```

Comment:
```suggestion
            onClick:
              options.trigger.value == "click"
                ? () => setExpanded(!options.isExpanded.value)
                : undefined,
```

We can omit the click handler all together if the trigger is not click :)